examples/recruting/views.py

```diff
@@ -57,3 +60,40 @@ def is_from_continent(  # pylint: disable=W0602, C0116, W9011
     @decorators.view_filter()
     def studied_at(self, university: str) -> sqlalchemy.ColumnElement:  # pylint: disable=W0602, C0116, W9011
         return Candidate.university == university
+
+
+class FewShotRecruitmentView(RecruitmentView):
+    """
+    A view for the recruitment database including examples of question:answers pairs (few-shot).
+    """
+
+    @decorators.view_filter()
+    def is_available_within_months(  # pylint: disable=W0602, C0116, W9011
+        self, months: int
+    ) -> sqlalchemy.ColumnElement:
+        start = date.today()
+        end = start + relativedelta(months=months)
+        return Candidate.available_from.between(start, end)
+
+    @decorators.few_shot()
+    def get_examples(self) -> List[FewShotExample]:  # pylint: disable=W0602, C0116, W9011
```

Comment:
let's add `question: str` as an argument here - it will allow the user to select / rerank few-shot examples based on similarity